themes/bootstrap3/templates/upgrade/confirmdeprecatedcolumns.phtml

```diff
@@ -0,0 +1,19 @@
+<?php $this->headTitle($this->translate('Deprecated database columns detected')) ?>
+
+<h2>Deprecated database columns detected</h2>
+
+<div class="alert alert-info">
+	Your database contains one or more columns that are no longer used, and which should not contain any useful information.
+</div>
+
+<table class="table table-striped">
+    <tr><th scope="col">Table</th><th scope="col">Column(s)</th></tr>
+    <?php foreach ($deprecated as $table => $columns): ?>
+        <tr><td><?=$this->escapeHtml($table)?></td><td><?=$this->escapeHtml(implode(', ', $columns))?></td></tr>
+    <?php endforeach; ?>
+</table>
+
+<p><b>RECOMMENDED:</b> You can <a href="<?=$this->url('upgrade-confirmdeprecatedcolumns')?>?action=delete">remove the columns</a>.</p>
+
+<p>Alternatively, if you are using these columns for some reason (e.g. in local custom code), you can choose to <a href="<?=$this->url('upgrade-confirmdeprecatedcolumns')?>?action=keep">keep the columns</a>.</p>
```

Comment:
As regular link, it looks like it will take you to documentation on how to do this. I would instead style these links as buttons by adding the classes `btn btn-default`.

Reply:
Thanks, @crhallberg. I had to use `btn btn-default btn-link` to avoid having underlines inside the buttons. Is that appropriate, or should I do things differently? (If I leave off `btn-default` they don't look like buttons).

Reply:
That makes sense, I'm surprised that `.btn-default` doesn't remove the underline. That might be a more consistent option.

Reply:
Same applies to #2741.

Reply:
Please take a look at my latest changes to #2741. If you approve of that approach, I can update this to match once it is merged.

Reply:
Okay, @crhallberg, #2741 has been merged into this, and I've adjusted the template accordingly. I think this is ready to merge now; just needs your approval.